src/components/WorkSchedule/ColumnRender.js

```diff
@@ -11,7 +11,8 @@ class ColumnRender extends Component {
     var result = data.getDishByWeek(this.props.randomWeek)[0]["recipes"][
       this.props.index + 1
     ];
-    return rendererFunc(result, this.props.mode);
+
+    return rendererFunc(result, this.props.mode, this.props.randomRecipes[this.props.index]);
```

Comment:
this function only making more and more complexier for me

Reply:
How can I make it easier?

Reply:
it's a hard question. First of all - i can guess that this function is heavy - so the easiest way - to split it into a few functions. Second thing - we have a 3 params, and they are long. i mean - i can bet that we have a lot of similar data inside this params. let's make this a separated task.

Reply:
let's skip it for now - buzz me when you fix my comments to this PR so i can merge it